app/commands/payments/stripe/payment/create.rb

```diff
@@ -14,27 +14,38 @@ def initialize
   initialize_with :user, :stripe_data, subscription: nil
 
   def call
-    Donations::Payment::Create.(user, :stripe, external_id, amount_in_cents, external_receipt_url, subscription:)
+    Payments::Payment::Create.(user, :stripe, product, external_id, amount_in_cents, external_receipt_url, subscription:)
   end
 
   private
   def external_id = stripe_data.id
   def external_receipt_url = stripe_data.charges.first.receipt_url
   def amount_in_cents = stripe_data.amount
 
+  def product
+    # TODO: is a payment intent guaranteed to have an invoice?
+    return :donation unless invoice
```

Comment:
It looks like a one-off payment does not have an invoice. After some googling, it looks like linking a product to a paymentintent is not supported, unless we pass in some metadata ourselves